lib/datasource/cdnjs/types.ts

```diff
@@ -1,14 +1,42 @@
+import { z } from 'zod';
+
 interface CdnjsAsset {
   version: string;
   files: string[];
-  sri?: Record<string, string>;
+  sri: Record<string, string>;
 }
 
-export interface CdnjsResponse {
+interface CdnjsResponse {
   homepage?: string;
   repository?: {
-    type: 'git' | unknown;
     url?: string;
   };
-  assets?: CdnjsAsset[];
+  assets: CdnjsAsset[];
+}
+
+/**
+ * See: https://cdnjs.com/api
+ */
+export const Response = z.object({
+  homepage: z.string().url().optional(),
+  repository: z
+    .object({
+      url: z.string().url().optional(),
+    })
+    .optional()
+    .nullable(),
+  assets: z.array(
+    z.object({
+      version: z.string(),
+      files: z.array(z.string()),
+      sri: z.record(z.string()),
+    })
+  ),
+});
+
+// export type IResponse = z.infer<typeof Response>;
+export type IResponse = CdnjsResponse;
+
+export function validResponse(input: unknown): IResponse {
+  return Response.parse(input) as IResponse;
 }
```

Comment:
Move to `./common.ts`, `type.ts` files should only contain types, so they don't get loaded on runtime.